src/workerd/util/sqlite.c++

```diff
@@ -446,22 +477,46 @@ bool SqliteDatabase::isAuthorized(int actionCode,
       // We currently only permit a few pragmas.
       {
         kj::StringPtr pragma = KJ_ASSERT_NONNULL(param1);
+
         if (pragma == "table_list") {
           // Annoyingly, this will list internal tables. However, the existence of these tables
           // isn't really a secret, we just don't want people to access them.
           return true;
+          // TODO function_list & pragma_list should be authorized but return
+          // ALLOWED_SQLITE_FUNCTIONS & ALLOWED_[READ|WRITE]_PRAGMAS
+          // respectively
         } else if (pragma == "table_info") {
           // Allow if the specific named table is not protected.
-          KJ_IF_MAYBE(name, param2) {
+          KJ_IF_MAYBE (name, param2) {
             return regulator.isAllowedName(*name);
           } else {
-            return false;  // shouldn't happen?
+            return false; // shouldn't happen?
           }
-        } else if (pragma == "data_version") {
-          return true;
-        } else if (pragma == "foreign_keys") {
+        }
+
+        static const kj::HashSet<kj::StringPtr> allowedWritePragmas = []() {
+          kj::HashSet<kj::StringPtr> result;
+          for (const kj::StringPtr& func: ALLOWED_WRITE_PRAGMAS) {
+            result.insert(func);
+          }
+          return result;
+        }();
+
+        if (allowedWritePragmas.contains(pragma)) {
           return true;
         }
+
+        static const kj::HashSet<kj::StringPtr> allowedReadPragmas = []() {
```

Comment:
Instead of creating separate HashSets, maybe we want a `HashMap<StringPtr, Signature>`, where `Signature` is an enum of different argument signatures (NONE, BOOL, NAME)?